cmake/http-parser.cmake

```diff
@@ -14,6 +14,27 @@
 
 cmake_minimum_required(VERSION 2.8)
 
-set(HTTPPARSER_ROOT ${DEP_ROOT}/http-parser)
-set(HTTPPARSER_INCDIR ${HTTPPARSER_ROOT})
-set(HTTPPARSER_LIB ${LIB_ROOT}/libhttpparser.a)
+if("${TARGET_OS}" STREQUAL "NUTTX")
+  set(HTTPPARSER_NUTTX_ARG -DNUTTX_HOME=${TARGET_SYSTEMROOT})
+endif()
+
+set(DEPS_HTTPPARSER deps/http-parser)
+set(DEPS_HTTPPARSER_SRC ${ROOT_DIR}/${DEPS_HTTPPARSER}/)
+ExternalProject_Add(http-parser
+  PREFIX ${DEPS_HTTPPARSER}
+  SOURCE_DIR ${DEPS_HTTPPARSER_SRC}
+  BUILD_IN_SOURCE 0
+  BINARY_DIR ${DEPS_HTTPPARSER}
+  INSTALL_COMMAND ""
```

Comment:
If there is no INSTALL_COMMAND specified the external project step always tries to invoke the install part by default. There are multiple ways to disable the install part, this is one of them. However I've reworked this as for the nuttx build there were additional requirements.